plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageListenerConfig.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage.config;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.trino.plugin.openlineage.OpenLineageTransport;
+import io.trino.plugin.openlineage.OpenLineageTrinoFacet;
+import io.trino.spi.resourcegroups.QueryType;
+import jakarta.validation.constraints.NotNull;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class OpenLineageListenerConfig
+{
+    private OpenLineageTransport transport = OpenLineageTransport.CONSOLE;
+    private String trinoHost;
+    private Integer trinoPort;
+    private List<OpenLineageTrinoFacet> disabledFacets = Collections.emptyList();
+
+    private List<QueryType> includeQueryTypes = Arrays.asList(QueryType.DELETE,
+            QueryType.INSERT, QueryType.MERGE, QueryType.UPDATE, QueryType.ALTER_TABLE_EXECUTE);
+
+    @ConfigDescription("Type of transport used to emit lineage information.")
+    @Config("openlineage-event-listener.openlineage.transport.type")
+    public OpenLineageListenerConfig setTransport(OpenLineageTransport transport)
+    {
+        this.transport = transport;
+        return this;
+    }
+
+    public OpenLineageTransport getTransport()
+    {
+        return transport;
+    }
+
```

Comment:
I would stick to having this in config because server address might be changing (for example in kubernetes) and we don't get DNS name of trino coordinator anywhere in event metadata.